library/src/main/java/com/novoda/downloadmanager/LiteDownloadBatchStatus.java

```diff
@@ -105,26 +120,43 @@ public void markAsDeleted() {
         notificationSeen = false;
     }
 
+    @WorkerThread
     @Override
     public void markAsError(Optional<DownloadError> downloadError, DownloadsBatchStatusPersistence persistence) {
         this.status = Status.ERROR;
         this.downloadError = downloadError;
-        updateStatus(status, persistence);
+        updateStatusAsync(status, persistence);
     }
 
+    @WorkerThread
```

Comment:
Why does this one need to be on a `workerThread` if it immediately goes into `Async`? Same with the others here.

Reply:
Some of them do not apply anymore, on it